cdap-app-templates/cdap-etl/hydrator-test/src/main/java/io/cdap/cdap/etl/mock/transform/RecoveringTransform.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.mock.transform;
+
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.plugin.PluginClass;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.Transform;
+import io.cdap.cdap.etl.proto.v2.ETLPlugin;
+
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Plugin(type = Transform.PLUGIN_TYPE)
+@Name(RecoveringTransform.NAME)
+/**
+ * Transform that throws exception for first few calls and then recovers.
+ */
+public class RecoveringTransform extends Transform<StructuredRecord, StructuredRecord> {
+
+  public static final PluginClass PLUGIN_CLASS = getPluginClass();
+  public static final String NAME = "RecoveringTransform";
+  private static AtomicInteger exceptionCounter = new AtomicInteger(0);
```

Comment:
is there a reason this needs to be static and reset? Seems it would be simpler to just create this for each instance of the transform, then no reset is needed.

Reply:
Keeping in static as discussed offline.